Framework/PythonInterface/setup.py

```diff
@@ -6,15 +6,14 @@
 #     & Institut Laue - Langevin
 # SPDX - License - Identifier: GPL - 3.0 +
 #  This file is part of the mantid workbench.
+import os
 from setuptools import find_packages, setup
 
-@SETUPTOOLS_BUILD_COMMANDS_DEF@
 
-# The most basic setup possible to be able to use setup.py develop
+# The most basic setup possible to be able to use pip develop/install
 setup(
     name='mantid',
-    version='@VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_PATCH@@VERSION_TWEAK_PY@',
+    version=os.environ['MANTID_VERSION_STR'],
```

Comment:
So I agree it does mean that this variable now needs to be set in the environment but I found that when I looked back at the cmake side of things it was just as hard to understand how the `setup.py.in` templates fit together with the custom build rule and cmake targets that overall I opted for simplifying the thing to have a standard `setup.py` with the sacrifice that we get an error if the variable is not set.
At this point we've only ever driven this with cmake and if something changes it is probably best to evaluate it at the time rather than second guessing how we might use these things in the future?

Reply:
:+1: that's a good point especially since someone has to dig out the VERSION_X variables
I agree about not speculating it, I assumed that some external tooling could run this rather than it being always invoked by CMake